packages/cli/src/credentials/credentials-finder.service.ts

```diff
@@ -18,6 +18,37 @@ export class CredentialsFinderService {
 		private readonly roleService: RoleService,
 	) {}
 
+	/**
+	 * Fetches global credentials from the database.
+	 */
+	private async fetchGlobalCredentials(trx?: EntityManager): Promise<CredentialsEntity[]> {
+		const em = trx ?? this.credentialsRepository.manager;
+		return await em.find(CredentialsEntity, {
+			where: { isGlobal: true },
+			relations: { shared: true },
+		});
+	}
+
+	/**
+	 * Merges global credentials with the provided credentials list,
+	 * deduplicating based on credential ID.
+	 */
+	private mergeAndDeduplicateCredentials<T extends { id: string }>(
+		credentials: T[],
+		globalCredentials: CredentialsEntity[],
+		mapGlobalCredential: (cred: CredentialsEntity) => T | null,
+	): void {
+		const credentialMap = new Map(credentials.map((c) => [c.id, c]));
```

Comment:
Minor: Set of IDs would probably fit this better as we never use the values for anything

Reply:
addressed. thanks for the feedback